CMakeLists.txt

```diff
@@ -1 +1,27 @@
-add_subdirectory(src)
+# Project meant to be used as part of a super project including libutil and libendian
+cmake_minimum_required(VERSION 3.8)
+project(libsiedler2)
+
+include(RttrBoostCfg)
+find_package(Boost 1.64.0 REQUIRED COMPONENTS system filesystem iostreams)
+
+file(GLOB _sources src/*.cpp)
+file(GLOB_RECURSE _headers include/*.h include/*.hpp)
```

Comment:
ich hätte die eigentliche Bibliothek trotzdem lieber in in einem CMakeLists.txt in src

Reply:
Hatte das an anderen Stellen auch so, ist aber sehr unschön am Ende. Hatte mich dann erkundigt, was andere so machen und wie es sinnvoll wäre. Und am Ende kam das hier raus: `CMakeLists.txt +src+include+tests` in einer Ebene. Die lib wird in der Haupt-CMake erstellt, die tests extra. Mehr lohnt nicht und macht es unnötig umständlich (parent dir holen z.B.). Und wozu eine CMakeLists wo (fast) nur `add_subdir(src)` drin ist?

Reply:
Hab ich glaube gerade gemacht ;) Jenkinsfile sehe ich keine notwendigkeit. Alle Fehler sollten von travis abgefangen werden und jenkins checkt dann den komplett build später nochmal